packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/models/AnnotationInfoSourceModel.java

```diff
@@ -24,8 +25,11 @@ public String getName() {
     }
 
     @Override
-    protected ClassInfoModel prepareClassInfo() {
-        return ClassInfoModel.of(origin.getClassInfo());
+    protected Optional<ClassInfoModel> prepareClassInfo() {
+        var cls = origin.getClassInfo();
+
+        return cls != null ? Optional.of(ClassInfoModel.of(cls))
+                : Optional.empty();
```

Comment:
Could use `Optional.map()`:

```suggestion
        return Optional.of(origin.getClassInfo()).map(ClassInfoModel::of);
```

Just nitpicking here, feel free to ignore.

Reply:
Yeah, I thought about that and chose the ternary just because it has one time less object creation than the `Optional.map`. Guess it is kind of a micro optimization though.